src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,34 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate localDateFrom = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate localDateTo = LocalDate.parse(dateTo, FORMATTER);
+        String[] salaryData;
+        StringBuilder result = new StringBuilder().append("Report for period ")
+                .append(dateFrom).append(" - ").append(dateTo);
+
+        for (String name : names) {
+            int salary = 0;
+            for (String currentLine : data) {
+                salaryData = currentLine.split(" ");
+                LocalDate currentDate = LocalDate.parse(salaryData[0], FORMATTER);
+                String currentName = salaryData[1];
+                int workingHours = Integer.parseInt(salaryData[2]);
+                int incomePerHour = Integer.parseInt(salaryData[3]);
```

Comment:
[checklist](https://mate-academy.github.io/jv-program-common-mistakes/java-core/exceptions/salary-info): Any magic numbers should be constants
Your code should be easy to read. Move all hardcoded values to [constant fields](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names) and give them informative names.
Lets make numbers 0, 1, 2, 3 as constants with informative names